Source/Public/LevelStatsCollector.h

```diff
@@ -42,9 +62,24 @@ class MAPMETRICSGENERATION_API ALevelStatsCollector final : public AActor
         float GroundHeight;
     };
 
+    enum class ECaptureState
```

Comment:
i would create a small state machine with a small class hierarchy because you already have quite some code that would benefit of a more structured approach
you would `Enter`, `Tick`  and `Exit` virtual functions
each time you start a new phase you create the derived object

Reply:
will do tomorrow :D

Reply:
should be fine now